src/main/scala/org/scalacheck/Gen.scala

```diff
@@ -803,6 +812,12 @@ object Gen extends GenArities with GenVersionSpecific {
   def asciiPrintableStr: Gen[String] =
     listOf(asciiPrintableChar).map(_.mkString)
 
+  /** Generates a string that can represent a valid hexadecimal digit. This
+    * includes both upper and lower case values.
+    */
+  def hexStr: Gen[String] =
+    listOf(hexChar).map(_.mkString)
+
```

Comment:
This produces a lengthy string by default (up to 100?).  That works for your test with `BigInteger`, but may not for when people want byte, int or long from hex?

I guess you've copied the definition of `Gen.numStr`.  I guess it has these same sort of problems.

Reply:
Again, I agree. I did this for consistency.  That being said, I'm not too worried about this. If people want to create a `String` which represents a bounded type, such as `i32`, then it would be consistent with current idioms to just map on this generator.